QUARKUS-2391.md

```diff
@@ -0,0 +1,38 @@
+# QUARKUS-2391 Promote the OpenTelemetry extension from TP to full support
+JIRA:https://issues.redhat.com/browse/QUARKUS-2391
+## Scope of the testing
+Quarkus already supported Opentelemetry as a tech preview, but now we move towards full support, then we will extend our test coverage.
+### Existing tests
+In the current quarkus-test-suite exist two modules dedicated to opentelemetry in the monitoring folder, they are:
+
+- opentelemetry [monitoring-opentelemetry]
+- opentelemetry-reactive [monitoring-opentelemetry-reactive]
+
+These tests utilize trace export to Jaeger (all-in-one pod) components and context propagation through a Ping-Pong application with 3 pods (ping service,pong service, jaeger-all-in-one).
+The traces are directly sent to the Jaeger collector, and they also test Quarkus gRPC communication between services and SSE (Server-Sent Events).
+
+On the other hand, also other modules use ```quarkus-opentelemetry``` dependency:
+- javaee-like-getting-started
+- build-time-analytics
+- http/graphql-telemetry
+- http/vertx-web-client
+- sql-db/narayana-transactions (where is tested JDBC Opentelemetry instrumentation)
+
+#### Test scenarios
+- Disable Opentelemetry extensions to ensure the application behaves as expected without tracing enabled.
+- Testing scenario with additional configuration options such as Id Generator, Propagators, Resource, Sampler configuration.
+- Test with other exporters such a Zipkin.
+- Integration tests to ensure proper tracing is enabled and functioning correctly in other core Quarkus extensions such as ```quarkus-smallrye-reactive-messaging-amqp```.
```

Comment:
Also, we recently dropped opentracing, so we can try to move grpc coverage to opentelemetry as well as well: https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-3660.md?plain=1#L17

Reply:
Ok, as far as I know, in the current monitoring-opentelemetry modules tests we already have some gRPC coverage integrated here: 
https://github.com/quarkus-qe/quarkus-test-suite/blob/85a50304c23d39a264eb0530872c57560ceb2c02/monitoring/opentelemetry/pom.xml#L36

https://github.com/quarkus-qe/quarkus-test-suite/blob/85a50304c23d39a264eb0530872c57560ceb2c02/monitoring/opentelemetry-reactive/pom.xml#L32

But not a full module dedicated as you mentioned in that test plan: 
https://github.com/quarkus-qe/quarkus-test-suite/tree/main/monitoring/opentracing-reactive-grpc

So, it's a good idea to include it, thanks.

Reply:
If we have the same coverage in `opentelementry-reactive`, we can left it here and just drop opentracing. The only important thing is to have coverage for grpc+opentelemetry, it doesn't matter, in which module.